rasa/shared/core/domain.py

```diff
@@ -252,195 +255,213 @@ def from_directory(cls, path: Text) -> "Domain":
+        # if not included in the original domain dict
+        if "config" not in data:
+            data.update(
+                {"config": {"store_entities_as_slots": store_entities_as_slots}}
+            )
 
-        return duplicates
+        if SESSION_CONFIG_KEY not in data:
+            data.update(
+                {
+                    SESSION_CONFIG_KEY: {
+                        SESSION_EXPIRATION_TIME_KEY: (
+                            session_config.session_expiration_time
+                        ),
+                        CARRY_OVER_SLOTS_KEY: session_config.carry_over_slots,
+                    }
+                }
+            )
 
-    @staticmethod
-    def merge_dicts(
-        tempDict1: Dict[Text, Any],
-        tempDict2: Dict[Text, Any],
-        override_existing_values: bool = False,
-    ) -> Dict[Text, Any]:
-        """Merges two dicts."""
-        if override_existing_values:
-            merged_dicts, b = tempDict1.copy(), tempDict2.copy()
+        # KEY_E2E_ACTIONS, KEY_ACTIONS AND KEY_RESPONSES are required
+        # because they get added through the chaining of importers
+        if KEY_E2E_ACTIONS not in data:
+            data.update({KEY_E2E_ACTIONS: action_texts})
 
-        else:
-            merged_dicts, b = tempDict2.copy(), tempDict1.copy()
-        merged_dicts.update(b)
-        return merged_dicts
+        if KEY_ACTIONS not in data:
+            data.update({KEY_ACTIONS: []})
 
-    @staticmethod
-    def merge_lists(list1: List[Any], list2: List[Any]) -> List[Any]:
-        """Merges two lists."""
-        return sorted(list(set(list1 + list2)))
+        if KEY_RESPONSES not in data:
+            data.update({KEY_RESPONSES: {}})
+
+        if "version" not in data:
+            data.update({"version": LATEST_TRAINING_DATA_FORMAT_VERSION})
+
+        return data
 
     @staticmethod
-    def merge_lists_of_dicts(
-        dict_list1: List[Dict],
-        dict_list2: List[Dict],
-        override_existing_values: bool = False,
-    ) -> List[Dict]:
-        """Merges two dict lists."""
-        dict1 = {
-            (sorted(list(i.keys()))[0] if isinstance(i, dict) else i): i
-            for i in dict_list1
-        }
-        dict2 = {
-            (sorted(list(i.keys()))[0] if isinstance(i, dict) else i): i
-            for i in dict_list2
-        }
-        merged_dicts = Domain.merge_dicts(dict1, dict2, override_existing_values)
-        return list(merged_dicts.values())
+    def _sanitize_intents_in_domain_dict(data: Dict) -> Dict:
+        for intent in data.get(KEY_INTENTS, []):
+            if isinstance(intent, dict):
+                for intent_name, intent_property in intent.items():
+                    try:
+                        if not intent_property[
+                            USE_ENTITIES_KEY
+                        ]:  # this covers False, None and []
+                            intent_property[USE_ENTITIES_KEY] = []
```

Comment:
Why do we have to do this? And what about `IGNORE_ENTITIES_KEY`?

Reply:
There are 2 unit tests that fail without this part in `test_domain`: `test_load_intents_from_as_dict_representation` which I've adapted, and `test_clean_domain_for_file`. I could add a similar check for `IGNORE_ENTITIES_KEY`, maybe it got missed from unit testing.